### PR TITLE
Integrate Chatwoot for User Support

### PR DESCRIPTION
**Pull request by sagararyal**

### CODE REVIEW

File: pages/profile/index.tsx [hunk 1]
```diff
@@ -24,6 +25,14 @@ function ProfilePage(): ReactElement {
       if (user) {
         setProfile(user);
         setAuthenticatedType('private');
+        window.$chatwoot.setCustomAttributes({
```
Comment (sagararyal) [inline pages/profile/index.tsx:28]: instrument.js:109 TypeError: Cannot read properties of undefined (reading 'setCustomAttributes')

Screen Shot 2022-01-04 at 3 00 15 PM

File: src/features/user/ChatwootWidget/index.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+
+class ChatwootWidget extends React.Component {
+  componentDidMount() {
+    // Add Chatwoot Settings
+    window.chatwootSettings = {
+      hideMessageBubble: false,
+      position: 'right', // This can be left or right
+      locale: 'en', // Language to be set
```
Comment (sagararyal) [inline src/features/user/ChatwootWidget/index.tsx:9]: --> `locale: user.locale ? user.locale : 'en',`

File: package-lock.json [hunk 3]
```diff
@@ -1,13722 +1,8 @@
 {
   "name": "planet-web2",
   "version": "0.2.0",
-  "lockfileVersion": 2,
+  "lockfileVersion": 1,
```
Comment (norbertschuler) [inline package-lock.json]: Please revert the change of this file. We are using lockfileVersion 2 now. Please upgrade your npm version to at least 7.x, better 8.x.

**Comment by sagararyal** [comment]
#### 📝 Changed routes:


| [**/profile**](https://planet-webapp-osaeuxud4-planetapp.vercel.app/profile) |
|:-:|
|<a href="https://planet-webapp-osaeuxud4-planetapp.vercel.app/profile">Screenshot of /profile</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/ala0-P4bTmFT_pn04dmZ3t0rmyqG4uN---DzdbFN3Cs54_co7CEv6Xfr067MedJj79Cu35XVdZ5y3Uej9fD4WA.png">(view full size)</a> |







Commit 9bc57f11d632e991d408076b07161db695279aa6 (https://planet-webapp-osaeuxud4-planetapp.vercel.app).

File: pages/profile/index.tsx (hunk 1 above)
Comment (norbertschuler) [reply pages/profile/index.tsx]: fixed

File: src/features/user/ChatwootWidget/index.tsx (hunk 2 above)
Comment (norbertschuler) [reply src/features/user/ChatwootWidget/index.tsx:9]: I localized the widget with `window.$chatwoot.setLocale(user.locale ? user.locale : 'en');`, but the remaining custom welcome message at the top of the chat window is still in English:

Bildschirmfoto 2022-01-04 um 15 31 55

**Review by norbertschuler — APPROVED** [review]
Works except having a non localized English welcome text.

File: src/features/user/ChatwootWidget/index.tsx (hunk 2 above)
Comment (sagararyal) [reply src/features/user/ChatwootWidget/index.tsx:9]: I think that's something we'll have to live with unless we use a different `websiteToken` for German.

Comment (norbertschuler) [reply src/features/user/ChatwootWidget/index.tsx]: using language stored for UI in localStorage

**Comment by sarveshpro** [comment]
Looking into alternative implementation
